ui/src/api/api.ts

```diff
@@ -103,6 +103,85 @@ export async function getVersion() {
   return result.build.version;
 }
 
+interface Tree {
+  [key: string]: any;
+}
+/**
+ * Fetch all metric values on one go using the list.
+ */
+export async function getMetricsAll() {
+  try {
+    const metrics = await getMetrics();
+    const paths = Object.keys(metrics);
+    // FIXME: path.forEach is more appropiate
+    await Promise.all(
+      paths.map(async (path) => {
+        const response = await getMetric(path);
+        metrics[path] = response;
+        return { path: response };
+      })
+    );
+
+    return metrics;
+  } catch (error) {
+    console.error("Error:", error);
+  }
+  return {};
+}
+
+/**
+ * Get list of metrics.
+ */
+async function getMetrics() {
+  const path = "/actuator/metrics";
+  return await get(path)
+    .then((data) => {
+      // Check if the data has 'names' property
+      if (data.hasOwnProperty("names")) {
+        // Initialize an empty tree
+        let tree: Tree = { _bare: {} };
+
+        // Process each name
+        data.names.forEach((name: string) => {
+          tree["_bare"][name] = {};
+        });
+
+        // console.log(tree);
+        return JSON.parse(JSON.stringify(tree["_bare"]));
+      } else {
+        console.log("No names found in the data");
+        return {};
+      }
+    })
+    .catch((error) => {
+      console.error(`Error fetching ${path}`, error);
```

Comment:
we want to be able to see errors in the ui, so they need to be passed through the api call

Reply:
Fixed